opm/material/densead/Evaluation.hpp

```diff
@@ -320,35 +329,47 @@ class Evaluation
 
     template <class RhsValueType>
     bool operator>(RhsValueType other) const
-    { return this->value > other; }
+    { return this->value_ > other; }
 
     bool operator>(const Evaluation& other) const
-    { return this->value > other.value; }
+    { return this->value_ > other.value_; }
 
     template <class RhsValueType>
     bool operator<(RhsValueType other) const
-    { return this->value < other; }
+    { return this->value_ < other; }
 
     bool operator<(const Evaluation& other) const
-    { return this->value < other.value; }
+    { return this->value_ < other.value_; }
 
     template <class RhsValueType>
     bool operator>=(RhsValueType other) const
-    { return this->value >= other; }
+    { return this->value_ >= other; }
 
     bool operator>=(const Evaluation& other) const
-    { return this->value >= other.value; }
+    { return this->value_ >= other.value_; }
 
     template <class RhsValueType>
     bool operator<=(RhsValueType other) const
-    { return this->value <= other; }
+    { return this->value_ <= other; }
 
     bool operator<=(const Evaluation& other) const
-    { return this->value <= other.value; }
+    { return this->value_ <= other.value_; }
+
+    ValueType value() const
+    { return value_; }
+
+    void setValue(const ValueType& val)
+    { value_ = val; }
+
+    ValueType derivative(unsigned varIdx) const
+    { return derivatives_[varIdx]; }
+
+    void setDerivative(unsigned varIdx, const ValueType& derVal)
+    { derivatives_[varIdx] = derVal; }
```

Comment:
Here, we could add `assert(varIdx < size)` before the assignment, to protect against misuse.